cert_renewal: Watch kubelet-serving signer along with kube-apiserver-client-kubelet

praveenkumar opened the pull request:
Since 4.15, user provisioned infra (UPI) is used for creating the
bundle, kubelet-serving signer csr request also need to be approved
manually as per docs. This PR watch kubelet-serving signer and check if
it pending then approve it.

- https://docs.openshift.com/container-platform/4.15/backup_and_restore/control_plane_backup_and_restore/disaster_recovery/scenario-3-expired-certs.html

praveenkumar committed:
cert_removal: Remove code path which is fixed in later version of OCP

https://bugzilla.redhat.com/show_bug.cgi?id=1997906 was fixed during 4.9
timeframe but we are still using this workaround. This PR remove this
workaround.

praveenkumar committed:
cert_renewal: Watch kubelet-serving signer along with kube-apiserver-client-kubelet

Since 4.15, user provisioned infra (UPI) is used for creating the
bundle, kubelet-serving signer csr request also need to be approved
manually as per docs. This PR watch kubelet-serving signer and check if
it pending then approve it.

- https://docs.openshift.com/container-platform/4.15/backup_and_restore/control_plane_backup_and_restore/disaster_recovery/scenario-3-expired-certs.html

praveenkumar committed:
cert_renewal: Increase waiting time from 8min to 10min

During cert renewal testing, found out that once
kube-apiserver-client-kubelet signer cert is approved, around 9-10 mins
it takes by cluster to generate kubelet-serving cert instead of 8 mins.
This PR is just increase that waiting time.

praveenkumar committed:
Integration: Fix marketplace test using workaround

marketplace test cases are failing because of OLM issue and it should be
fix soon but until it does this PR use the workaround for this issue.

Note: We will revert this commit once it is fixed.

- https://issues.redhat.com/browse/OCPBUGS-31645

praveenkumar commented:
/retest

anjannath commented:
Tested locally and working as expected.

```
$ crc start -p ~/Downloads/pull-secret -b ~/crc_libvirt_4.15.3_amd64.crcbundle
INFO Using bundle path /home/anjan/crc_libvirt_4.15.3_amd64.crcbundle
INFO Checking if running as non-root
INFO Checking if running inside WSL2
INFO Checking if crc-admin-helper executable is cached
INFO Checking if running on a supported CPU architecture
INFO Checking if crc executable symlink exists
INFO Checking minimum RAM requirements
INFO Check if Podman binary exists in: /home/anjan/.crc/bin/oc
INFO Checking if Virtualization is enabled
INFO Checking if KVM is enabled
INFO Checking if libvirt is installed
INFO Checking if user is part of libvirt group
INFO Checking if active user/process is currently part of the libvirt group
INFO Checking if libvirt daemon is running
INFO Checking if a supported libvirt version is installed
INFO Checking if crc-driver-libvirt is installed
INFO Checking crc daemon systemd socket units
INFO Checking if systemd-networkd is running
INFO Checking if NetworkManager is installed
INFO Checking if NetworkManager service is running
INFO Checking if dnsmasq configurations file exist for NetworkManager
INFO Checking if the systemd-resolved service is running
INFO Checking if /etc/NetworkManager/dispatcher.d/99-crc.sh exists
INFO Checking if libvirt 'crc' network is available
INFO Checking if libvirt 'crc' network is active
INFO Loading bundle: crc_libvirt_4.15.3_amd64...
INFO Creating CRC VM for OpenShift 4.15.3...
INFO Generating new SSH key pair...
INFO Generating new password for the kubeadmin user
INFO Starting CRC VM for openshift 4.15.3...
INFO CRC instance is running with IP 192.168.130.11
INFO CRC VM is running
INFO Updating authorized keys...
INFO Configuring shared directories
INFO Check internal and public DNS query...
INFO Check DNS query from host...
INFO Verifying validity of the kubelet certificates...
INFO Starting kubelet service
INFO Kubelet client certificate has expired, renewing it... [will take up to 10 minutes]
INFO Kubelet serving certificate has expired, waiting for automatic renewal... [will take up to 8 minutes]
INFO Waiting for kube-apiserver availability... [takes around 2min]
INFO Adding user's pull secret to the cluster...
INFO Updating SSH key to machine config resource...
INFO Waiting until the user's pull secret is written to the instance disk...
INFO Changing the password for the kubeadmin user
INFO Updating cluster ID...
INFO Updating root CA cert to admin-kubeconfig-client-ca configmap...
INFO Starting openshift instance... [waiting for the cluster to stabilize]
INFO 4 operators are progressing: authentication, image-registry, ingress, kube-apiserver
INFO 2 operators are progressing: image-registry, kube-apiserver
INFO 2 operators are progressing: image-registry, kube-apiserver
INFO 2 operators are progressing: image-registry, kube-apiserver
INFO Operator image-registry is progressing
INFO All operators are available. Ensuring stability...
INFO Operators are stable (2/3)...
INFO Operators are stable (3/3)...
INFO Adding crc-admin and crc-developer contexts to kubeconfig...
Started the OpenShift cluster.

The server is accessible via web console at:
  https://console-openshift-console.apps-crc.testing

Log in as administrator:
  Username: kubeadmin
  Password: xuRDv-RIK4U-kp46G-cDWYz

Log in as user:
  Username: developer
  Password: developer

Use the 'oc' command line interface:
  $ eval $(crc oc-env)
  $ oc login -u developer https://api.crc.testing:6443
```

anjannath commented:
this was from before and not directly related to this PR, but in line 77 there's `logging.Info("Kubelet serving certificate has expired, waiting for automatic renewal... [will take up to 8 minutes]")` which should be `5 minutes` as the retry timeout is 5mins

anjannath commented on test/e2e/features/story_openshift.feature:
```diff
@@ -70,6 +70,12 @@ Feature: 4 Openshift stories
 	@darwin @linux @windows @testdata @story_marketplace @cleanup @needs_namespace
 	Scenario: Install new operator
 		Given executing "oc new-project testproj" succeeds
+		 And executing "oc delete pods --all -n openshift-marketplace" succeeds
```
there's an extra white space before `And`

cfergeau commented:
Looks good to me!

praveenkumar committed:
cert_renewal: fix incorrect info for wait time in log info

Actually retry happen for 5 mins instead 8 mins. This PR fixes this
misinformation.
